solidity/test/bridge/Bridge.MovingFunds.test.ts

```diff
@@ -45,10 +49,501 @@ describe("Bridge - Moving funds", () => {
+        // by default. No need to do anything here.
+
+        context("when the commitment was not submitted yet", () => {
+          // The wallet created using the `walletDraft` has no commitment
+          // submitted by default. No need to do anything here.
+
+          context("when the caller is a member of the source wallet", () => {
+            const walletMembersIDs = [1, 2, 3, 4, 5]
+            const walletMemberIndex = 2
+
+            let caller: SignerWithAddress
+
+            before(async () => {
+              await createSnapshot()
+
+              caller = thirdParty
+
+              walletRegistry.isWalletMember
+                .whenCalledWith(
+                  ecdsaWalletTestData.walletID,
+                  walletMembersIDs,
+                  caller.address,
+                  walletMemberIndex
+                )
+                .returns(true)
+            })
+
+            after(async () => {
+              walletRegistry.isWalletMember.reset()
+
+              await restoreSnapshot()
+            })
+
+            context("when passed wallet main UTXO is valid", () => {
+              // Just an arbitrary main UTXO with value of 26 BTC.
+              const mainUtxo = {
+                txHash:
+                  "0xc9e58780c6c289c25ae1fe293f85a4db4d0af4f305172f2a1868ddd917458bdf",
+                txOutputIndex: 0,
+                txOutputValue: to1ePrecision(26, 8),
+              }
+
+              before(async () => {
+                await createSnapshot()
+
+                // Set up a main UTXO for the source wallet.
+                await bridge.setWalletMainUtxo(
+                  ecdsaWalletTestData.pubKeyHash160,
+                  mainUtxo
+                )
+              })
+
+              after(async () => {
+                await restoreSnapshot()
+              })
+
+              context(
+                "when the expected target wallets count is greater than zero",
+                () => {
+                  // Just some arbitrary 20-byte hashes to simulate live
+                  // wallets PKHs.
+                  const liveWallets = [
+                    "0xffb9e05013f5cd126915bc03d340cc5c1be81862",
+                    "0x4b440cb29c80c3f256212d8fdd4f2125366f3c91",
+                    "0x888f01315e0268bfa05d5e522f8d63f6824d9a96",
+                    "0xbf198e8fff0f90af01024153701da99b9bc08dc5",
+                    "0xb2a89e53a4227dbe530a52a1c419040735fa636c",
+                  ]
+
+                  before(async () => {
+                    await createSnapshot()
+
+                    for (let i = 0; i < liveWallets.length; i++) {
+                      // eslint-disable-next-line no-await-in-loop
+                      await bridge.setWallet(liveWallets[i], {
+                        ...walletDraft,
+                        state: walletState.Live,
+                      })
+                    }
+
+                    await bridge.setLiveWalletsCount(liveWallets.length)
```

Comment:
Non-blocking but we could consider increasing the number of live wallets as a part of the stub logic in `setWallet`. We are setting it separately everywhere and I think it's easy to forget for someone who is not that familiar with this code, so potentially error-prone.

Reply:
Good idea! See https://github.com/keep-network/tbtc-v2/pull/208/commits/1c90387f65f4ed8553231487f7c0efdc92bbcd3d